export/cloud_datastore_admin.py

```diff
@@ -12,107 +12,100 @@
-import json
 import logging
 
-from export import acl_decorators
+from core.controllers import base
 
 from google.appengine.api import app_identity
-from google.appengine.api import urlfetch
+import requests
 import webapp2
 
 APP_NAME_OPPIASERVER = 'oppiaserver'
+GCS_BUCKET_URL_PREFIX = 'gs://'
 
 
-class ExportToCloudDatastoreHandler(webapp2.RequestHandler):
+class ExportToCloudDatastoreHandler(base.BaseHandler):
     """Request handler which supports triggering automatic exports of the
     entities that application stores in Google Cloud Datastore.
     """
 
-    @acl_decorators.can_perform_cron_tasks
+    GET_HANDLER_ERROR_RETURN_TYPE = 'json'
+
+    @property
+    def is_cron_job(self):
+        """Returns whether the request came from a cron job."""
+        return self.request.headers.get('X-AppEngine-Cron') is not None
+
     def get(self):
         """Triggers an export of Google Cloud Datastore.
 
         Export data described in request parameters.
 
         Raises:
-            AssertionError. Bucket url exists and doesn't start with 'gs://'.
+            UnauthorizedUserException. The user does not have credentials to
+                access the page.
+            PageNotFoundException. The request was not made by the production
+                environment.
+            InvalidInputException. The bucket parameter is malformed or missing.
         """
-        gcs_bucket_url_prefix = 'gs://'
 
-        access_token, _ = app_identity.get_access_token(
-            'https://www.googleapis.com/auth/datastore')
-        app_id = app_identity.get_application_id()
+        if not (self.is_cron_job or self.is_super_admin):
+            e = self.UnauthorizedUserException(
+                'You do not have the credentials to access this page.')
+            logging.error(e)
+            raise e
 
+        app_id = app_identity.get_application_id()
         if app_id != APP_NAME_OPPIASERVER:
-            logging.error(
-                'Export service has been pinged. '
-                'Since this is not production, a real export request '
-                'has not been initiated.')
-            return
-
-        timestamp = datetime.datetime.utcnow().strftime('%Y%m%d-%H%M%S')
-
-        output_url_prefix = self.request.get('output_url_prefix')
-        assert output_url_prefix and output_url_prefix.startswith(
-            gcs_bucket_url_prefix)
-
-        # Look for slash in the portion of the bucket URL that comes
-        # after 'gs://'. If not present, then only a bucket name has been
-        # provided and we append a trailing slash.
-        if '/' not in output_url_prefix[len(gcs_bucket_url_prefix):]:
-             # Only a bucket name has been provided - no prefix or trailing
-             # slash.
-            output_url_prefix += '/' + timestamp
-        else:
-            output_url_prefix += timestamp
+            e = self.PageNotFoundException(
+                'Export service has been pinged from a non-production '
```

Comment:
It could be from the wrong app ID too, right?

Reply:
True, added "or non-Oppia" to address this.